### PR TITLE
Add delegated resource owners to ApiDepartment model

### PR DESCRIPTION
**Pull request by oivheg**
- [x] New feature
- [ ] Bug fix
- [ ] High impact

**Description of work:**
Updated department/{id} to default expand delegatedResourceOwners 
Updated apiProposedPerson and QueryProposedPerson
Expanding resoure onwer should now also expand delegatedResrouceOwners

**Testing:**
- [x] Can be tested
- [ ] Automatic tests created / updated
- [ ] ~~Local tests are passing~~

by verifying that the delegatedResourceOwners property has the desired data neccesary

**Checklist:**
- [ ] ~~Considered automated tests~~
- [ ] ~~Considered updating specification / documentation~~
- [x] Considered work items 
- [ ] ~~Considered security~~
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

### CODE REVIEW

**Comment by oivheg** [comment]
All tests appear to be running fine localy: 

image

**Comment by oivheg** [comment]
Den testen som feiler, fungere helt fin lokalt, og vi får delegatedResourseOwners. 
image

File: src/backend/api/Fusion.Resources.Api/Controllers/Requests/ApiModels/ApiProposedPerson.cs [hunk 1]
```diff
@@ -18,13 +20,17 @@ public ApiProposedPerson(QueryResourceAllocationRequest.QueryProposedPerson prop
...
+                DelegatedResourceOwners = new List<FusionPersonProfile?>(proposedPerson.DelegatedResourceOwners);
+            }
+        }
 
         public DateTimeOffset ProposedAt { get; set; }
         public ApiPerson Person { get; set; }
         public bool WasNotified { get; set; }
-
         public ApiPerson? ResourceOwner { get; set; }
+        public List<FusionPersonProfile?> DelegatedResourceOwners { get; set; }
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Controllers/Requests/ApiModels/ApiProposedPerson.cs:34]: Incorrect use of model.
Should be something like: 
public IEnumerable<ApiPerson>? DelegatedResourceOwners { get; set; }

File: src/backend/api/Fusion.Resources.Api/Controllers/Requests/ApiModels/ApiProposedPerson.cs [hunk 2]
```diff
@@ -18,13 +20,17 @@ public ApiProposedPerson(QueryResourceAllocationRequest.QueryProposedPerson prop
             {
                 ResourceOwner = new ApiPerson(proposedPerson.ResourceOwner);
             }
-        }
 
+            if (proposedPerson.DelegatedResourceOwners is not null)
+            {
+                DelegatedResourceOwners = new List<FusionPersonProfile?>(proposedPerson.DelegatedResourceOwners);
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Api/Controllers/Requests/ApiModels/ApiProposedPerson.cs:26]: given update of property, this must be changed to something like:
DelegatedResourceOwners = proposedPerson.DelegatedResourceOwners.Select(x => new ApiPerson(x));

File: src/backend/api/Fusion.Resources.Domain/Models/QueryProposedPerson.cs [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
 
-    //    public QueryPerson Person { get; set; }
-    //    public bool WasNotified { get; set; }
-    //}
-}
+        public FusionPersonProfile? Person { get; set; }
+
+        public FusionPersonProfile? ResourceOwner { get; set; }
+
+        public List<FusionPersonProfile?>? DelegatedResourceOwners { get; set; }
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Domain/Models/QueryProposedPerson.cs:17]: You shouldn't add NULL-objects to the list, but like this:

public IEnumerable<FusionPersonProfile>? DelegatedResourceOwners { get; set; }

File: src/backend/api/Fusion.Resources.Domain/Queries/GetResourceAllocationRequestItem.cs [hunk 4]
```diff
@@ -222,14 +223,25 @@ private async Task ExpandResourceOwnerAsync(QueryResourceAllocationRequest reque
                     if (request.ProposedPerson?.AzureUniqueId is not null)
                     {
                         var manager = await mediator.Send(new GetResourceOwner(request.ProposedPerson.AzureUniqueId));
-                        request.ProposedPerson.ResourceOwner = manager;
+
+                        if (manager != null)
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Domain/Queries/GetResourceAllocationRequestItem.cs:227]: manager?.FullDepartment is not null instead? 
A manager MAY not have FullDepartment populated

File: src/backend/api/Fusion.Resources.Domain/Queries/GetResourceAllocationRequestItem.cs [hunk 5]
```diff
@@ -222,14 +223,25 @@ private async Task ExpandResourceOwnerAsync(QueryResourceAllocationRequest reque
...
+                        {
+                            var department = await mediator.Send(new GetDepartment(manager.FullDepartment).ExpandDelegatedResourceOwners());
+
+                            request.ProposedPerson.ResourceOwner = manager;
+                            if (department != null)
+                            {
+                                request.ProposedPerson.DelegatedResourceOwners = department.DelegatedResourceOwners;
```
Comment (xssic) [inline src/backend/api/Fusion.Resources.Domain/Queries/GetResourceAllocationRequestItem.cs:234]: department may not exist?

department?.Delega.....

File: src/backend/tests/Fusion.Resources.Api.Tests/Helpers/Models/Responses/TestApiPerson.cs [hunk 6]
```diff
@@ -7,6 +7,7 @@ namespace Fusion.Testing.Mocks
     public class TestApiPerson
     {
         public Guid AzureUniquePersonId { get; set; }
+        public Guid AzureUniqueId { get; set; }
```
Comment (xssic) [inline src/backend/tests/Fusion.Resources.Api.Tests/Helpers/Models/Responses/TestApiPerson.cs:10]: Same as property above?

File: src/backend/tests/Fusion.Resources.Api.Tests/Helpers/Models/Responses/TestApiProposedPerson.cs [hunk 7]
```diff
@@ -9,7 +12,7 @@ public class TestApiProposedPerson
         public DateTimeOffset ProposedAt { get; set; }
         public TestApiPerson Person { get; set; } = null!;
         public bool WasNotified { get; set; }
-    }
-
 
-}
+        public List<TestApiPerson?>? DelegatedResourceOwners { get; set; }
```
Comment (xssic) [inline src/backend/tests/Fusion.Resources.Api.Tests/Helpers/Models/Responses/TestApiProposedPerson.cs:16]: List may be null, but shouldn't contain a null person

**Review by xssic — COMMENTED** [review]
Double check my comments, and make sure the build-warning's are fixed.
Then re-request review

File: src/backend/tests/Fusion.Resources.Api.Tests/Helpers/Models/Responses/TestApiProposedPerson.cs [hunk 8]
```diff
@@ -1,4 +1,7 @@
-﻿using System;
+﻿using Fusion.Integration.Profile;
```
Comment (xssic) [inline src/backend/tests/Fusion.Resources.Api.Tests/Helpers/Models/Responses/TestApiProposedPerson.cs:1]: remove unused usings.

File: src/backend/tests/Fusion.Resources.Api.Tests/IntegrationTests/DepartmentsController.cs [hunk 9]
```diff
@@ -6,6 +6,7 @@
 using Fusion.Testing;
 using Fusion.Testing.Mocks.LineOrgService;
 using Fusion.Testing.Mocks.OrgService;
+using SixLabors.ImageSharp.ColorSpaces;
```
Comment (xssic) [inline src/backend/tests/Fusion.Resources.Api.Tests/IntegrationTests/DepartmentsController.cs:9]: ??

File: src/backend/tests/Fusion.Resources.Api.Tests/IntegrationTests/DepartmentsController.cs [hunk 10]
```diff
@@ -80,10 +81,9 @@ public async Task SearchDepartment_Should_GetFromLineOrg_WhenNotInDb()
         [Fact]
         public async Task SearchDepartment_Should_GetDelegatedResponsibles_FromRoleService()
         {
-            //var department = "TPD LIN ORG TST1";
-            var delegatedDepartment = "TPD LIN ORG TST1";
+            var delegatedDepartment = "FromRole";
             var nonDelegatedDepartment = "Non delegated";
-
+            var Source = $"Soruce ID";
```
Comment (xssic) [inline src/backend/tests/Fusion.Resources.Api.Tests/IntegrationTests/DepartmentsController.cs:86]: ..

File: src/backend/tests/Fusion.Testing.Core/Http/TestClientHttpResponse.cs [hunk 11]
```diff
@@ -29,6 +29,9 @@ public TestClientHttpResponse(HttpResponseMessage response)
 
         public static async Task<TestClientHttpResponse<T>> CreateResponseAsync<T>(HttpResponseMessage response, bool skipDeserialization = false)
         {
+            var jsonSerializerSettings = new JsonSerializerSettings();
+            jsonSerializerSettings.MissingMemberHandling = MissingMemberHandling.Ignore;
```
Comment (xssic) [inline src/backend/tests/Fusion.Testing.Core/Http/TestClientHttpResponse.cs:33]: hmmm. shouldn't be required.

File: src/backend/tests/Fusion.Resources.Api.Tests/IntegrationTests/InternalResourceAllocationRequestTests.cs [hunk 12]
```diff
@@ -1,28 +1,43 @@
...
+using Fusion.Resources.Domain;
 using Fusion.Testing;
 using Fusion.Testing.Authentication.User;
 using Fusion.Testing.Mocks;
 using Fusion.Testing.Mocks.LineOrgService;
 using Fusion.Testing.Mocks.OrgService;
+using Fusion.Testing.Mocks.OrgService.Api;
 using Fusion.Testing.Mocks.ProfileService;
+using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Identity.Client;
```
Comment (xssic) [inline src/backend/tests/Fusion.Resources.Api.Tests/IntegrationTests/InternalResourceAllocationRequestTests.cs:25]: a lot of unknown usings.

**Review by xssic — COMMENTED** [review]
Please disable your auto-formatting tool.
A lot of files changed, with irrelevant changes.

**Comment by oivheg** [comment]
Autoformating tool, have been disabeled

**Review by xssic — APPROVED** [review]

**Pull request merged**